LSP-volar.sublime-settings

```diff
@@ -72,5 +72,8 @@
 			"scopes": ["text.html.vue"],
 			"syntaxes": ["Packages/Vue Syntax Highlight/Vue Component.sublime-syntax"],
 		}
-	]
+	],
+	"disabled_capabilities": {
+		"codeLensProvider": true
+	}
```

Comment:
Doesn't volar provide `languageFeatures.codeLens` or something.

Reply:
Yes, but languageFeatures are not exposed to the user.
I would still like to allow the user to enable code lens if they want.

Reply:
Are you sure those are not exposed. I think those will be read from `initializationOptions.languageFeatures.codeLens`

Reply:
You are right, yes, languageFeatures are part of the initializationOptions,
and yes the user could provide those.

But even if the user provides them,
the LSP-volar plugin is explicitly overriding them in is_allowed_to_start
https://github.com/sublimelsp/LSP-volar/blob/master/plugin.py#L35

And by that I meant that they are not exposed.

I believe that languageFeatures are not meant to be edited by end users.
Volar (in VS code and in coc-volar) starts 3 different processes with different languageFeatures enabled to provide responses faster. 
https://github.com/johnsoncodehk/volar/blob/master/extensions/vscode-vue-language-features/src/common.ts#L240

I am also thinking of providing multi server support in LSP-volar, 
but I stumbled upon this issue so I removed that functionality for the time being #116 :)

Reply:
Would then it maybe be preferred  to create a new configuration option in the usual place (the `settings` object).

Also it would be a bit awkward to override `disable_capabilities` in user settings. You'd have to create an empty `disable_capabilities` object or set `"codeLensProvider": false`? Not sure either of those would actually work...

I'd suggest that you create an `settings.enableCodeLens` settings (not in `volar` namespace on purpose so that the server might even ignore it) and set initializationOptions appropriately based on it.

Reply:
I could implement it as a separate `enableCodeLens` setting,
but isn't `disable_capabilities` LSP way of disabling capabilities?

Reply:
I feel like it's fragmenting the settings landscape for the user. Now user has to look in two different places for available settings.

But also, as I've mentioned, I think it would be awkward or maybe even impossible to override it.

Though, from another point of view, if we would allow to change `settings` without restarting the server then having it in `settings` would not apply the change...

Reply:
I'm unsure if I would add a new setting, I could, but I am not sure.

For me I just wanted to disable the code lens by default (rather to let them enabled by default), 
because I find them a bit distracting.

Rafal if you have a suggestion I will went with your proposal,
If you do not have a strong opinion(suggestion) of what the to do here, 
I will close this this PR and address it when someone reports it as issue. (I will locally just for myself disable the code lens capability)

Reply:
If you can verify that it's possible to override disabled capability from user settings then go for it

Reply:
> If you can verify that it's possible to override disabled capability from user settings then go for it

It is possible, 
but I got thinking that we should disable code lens by default in LSP.
So I will close this PR.